### ISSUE TITLE
Embodied Carbon Needs Detail

### ISSUE THREAD

**Issue by buchananwp** [Green-Software-Foundation/software_carbon_intensity#31]
Flagging this to work with @arexub (Alex Bitukov) from MSFT around the latest embodied work.

**Comment by atg-abhishek** [comment]
Would be great to have some more details from you @arexub so that we can bring it up for discussion once we have a bit more substantial detail and then explore how we can account for that in the SCI.

**Comment by jawache** [comment]
This is a rabbit hole, you can spend 30 mins on this or a whole year :) I think we need to **insist** on a very simple method that everyone can do and **encourage** a more involved method that only some orgs will be able to invest in.

Level one of this I believe should be cost-based, most people use a cost-based multiplier most carbon accounting tooling uses a cost-based multiplier. They are directionally accurate and fast to calculate. So for the cloud, you look at your cloud bill and multiple by a number, something that one person could calculate in 30 mins.

Level two we might explore a more involved approach where people figure out the exact specs of the hardware they used and reference that against a database of emissions factors.  That gives you a more accurate number but takes a lot longer and those emissions factor databases sometimes cost a lot of money.

**Comment by atg-abhishek** [comment]
@buchananwp can transfer to @arexub once he joins?

**Comment by jawache** [comment]
I was reminded of this article by Benjamin Davy: https://medium.com/teads-engineering/evaluating-the-carbon-footprint-of-a-software-platform-hosted-in-the-cloud-e716e14e060c

He goes into a lot of detail about how to calculate the embodied carbon emissions. To cut a long story short:

1. Use cost as a proxy, it's fast and is directionally accurate at least. A simple solution here would be to create a single coefficient carbon per $. Some sort of global average, I imagine several carbon accounting services have already calculated this since they give you a carbon number simply by looking at the cost of your cloud bills. There are more accurate methods, but this would be fastest.

2. Use emission factor databases there is a lack of these databases for digital products and as far as I know all the good ones cost money or are in-house. 

I think we need to branch off an innovation projects to calculate the embodied carbon coefficients of hardware which the GSF publishes for free. The SCI can reference our free database in the spec.

**Comment by atg-abhishek** [comment]
This is a great article @jawache and I think we can definitely take some cues from it and based on the discussion today, we can add some details about this in our SCI spec.

**Comment by buchananwp** [comment]
- [ ] incorporate lifespan (e.g. how do we amortize embodied carbon?) From our studies, it's shown that lifespan can be highly variable and hard to estimate.

**Comment by atg-abhishek** [comment]
Would the preset list that we have in the SCI be a good venue for this @navveenb ?

**Comment by vaughanknight** [comment]
@navveenb Agree on cost not being a good measure.  Cloud costs or on prem costs can vary based simply on exchange rate, and internal business factors, price increases and competitor pressure.  The costs of items in a data centre, or the data centre itself, can be a lot more expensive because of levels of physical security, legislation, downside risk etc, and have little to do with the hardware itself.

**Comment by vaughanknight** [comment]
> @navveenb @jawache "I think we need to branch off an innovation projects to calculate the embodied carbon coefficients of hardware which the GSF publishes for free. The SCI can reference our free database in the spec." - This is a good project to have for the community.

100% agree.  We should also have guidance on the site as to other references that exist.  Some already do exist but this will evolve over time as the world gets a bit wiser in relation to supply chain and understands the full emissions impact.

@buchananwp One area we need to make note of is I assume (and this is a big assumption) that carbon offsets in the supply chain of the hardware is included in SCI for embodied carbon, but not the companies own emissions, and not for the software itself.

**Comment by navveenb** [comment]
> preset list

@atg-abhishek  can you point me to the preset list values, i only see a place holder in the spec

**Comment by atg-abhishek** [comment]
@vaughanknight re: 
> @buchananwp One area we need to make note of is I assume (and this is a big assumption) that carbon offsets in the supply chain of the hardware is included in SCI for embodied carbon, but not the companies own hardware emissions (building, transport, setting up), and not for the software itself.

I don't think we want to include the offsets in any part of the SCI calculation unless I'm mistaken on what this is referring to?

**Comment by atg-abhishek** [comment]
@navveenb yes we only have a placeholder for now and it needs to be populated

**Comment by jawache** [comment]
@navveenb After seeing some of the work by Benjamin Davy I'm now leaning more towards a very simple model for estimating embodied emissions based on some coefficients for high level components rather than cost so happy to move away from cost as a proxy.

I'm thinking more and more that to support the SCI the Innovation WG needs to launch the "SCI Open Data" project, to make all the data needed to execute the SCI, including rough emission factors, freely available.

**Comment by atg-abhishek** [comment]
Agreed on this!

**Comment by navveenb** [comment]
> @navveenb After seeing some of the work by Benjamin Davy I'm now leaning more towards a very simple model for estimating embodied emissions based on some coefficients for high level components rather than cost so happy to move away from cost as a proxy.
> 
> I'm thinking more and more that to support the SCI the Innovation WG needs to launch the "SCI Open Data" project, to make all the data needed to execute the SCI, including rough emission factors, freely available.

Agreed @jawache , we should stay away from cost as it could be misleading..  I saw some of the inputs for SCI Open Data as part of SCI Core Library innovation project, which can be extended and make this data available for all.

**Comment by atg-abhishek** [comment]
reminder for @jawache @Henry-WattTime and others to add in thoughts based on the call from last week with Benjamin Davy. Thanks!

**Comment by Henry-WattTime** [comment]
Based on the presentation from Benjamin, I think we landed on:

(Total embodied emissions/expected lifespan * reserved usage)/(system utilization)

where:
 total embodied emissions is the sum of LCA emissions for all components
Expected lifespan is the anticipated time that the equipment will be installed 
reserved usage is the time the equipment has been reserved for use of the software
and system utilization is the anticipated total time the equipment is reserved divided by the total installed lifespan (if available). (Could be datacenter utilization percentage or something similar, like the percentage of time that equipment is reserved in the datacenter)

Is this recollection correct? Are the bones right? Any modifications?

**Comment by jawache** [comment]
Embodied carbon (otherwise referred to as "Embedded Carbon") is the amount of carbon pollution emitted during the creation and disposal of a device. They are a significant component of the total carbon emissions of software and therefore MUST be taken into account when calculating the SCI score of an application.

You MUST include all the embodied emissions for the hardware used within the boundary of your software. 

When using shared resources your share of the embodied carbon MUST be based on how much you have been allocated rather than your utilisation of the shared resource. (NOTE: It's about how much of the shared resource you are stopping someone else from using, i.e. if you buy a car and only you can use it, all of the embodied carbon is yours).

You MAY use simple models to estimate embodied emissions however we RECOMMEND where possible to use the most granular data possible when calculating your embodied carbon.

**Comment by jawache** [comment]
Some phrasing for discussion:

You MAY use simple models to estimate embodied emissions; however, we RECOMMEND using the most granular data possible and ideally emissions data from a devices life cycle analysis when calculating your embodied carbon.

And

You MUST include an estimate of all the embodied emissions for the hardware used within your software boundary.

**Comment by jawache** [comment]
Actually perhaps more like this `M = Total embodied emissions * time reserved/expected lifespan * reserved usage`

So for a server that has 40vCPUs that you rented 4vCPUs for 1 hr and has 4000kg of embodied emissions assuming a 4 year lifespan.

`M = 4000 * 1h/4years * 4/40`

For a laptop, you might choose a different way to calculate `reserved usage`? Laptops are usually limited by memory in my experience, you don't start killing apps until you need to free up mem.

So for a laptop that has 16gb RAM and 1000kg of embodied emissions assuming a 5 year lifespan, you're app ran for 20 mins and took up 4gb RAM.

`M = 1000 * 20mins/5years * 4/16`

**Comment by atg-abhishek** [comment]
> You MAY use simple models to estimate embodied emissions; however, we RECOMMEND using the most granular data possible and ideally emissions data from a devices life cycle analysis when calculating your embodied carbon.

Since `RECOMMEND` isn't on [RFC2119](https://datatracker.ietf.org/doc/html/rfc2119), maybe we can rephrase this as follows: (`RECOMMENDED` is in RFC2119, but I'd rather keep everything in the sentence for the same subject, i.e. "you")

> You MAY use simple models to estimate embodied emissions; however, you SHOULD use the most granular data possible and ideally emissions data from a devices life cycle analysis when calculating your embodied carbon.

**Comment by atg-abhishek** [comment]
thanks @Henry-WattTime and @jawache for capturing this here :)

**Comment by atg-abhishek** [comment]
Sounds good @Sealjay - thanks!

**Comment by buchananwp** [comment]
@tgogoi for context
